### DISCUSSION TITLE
How to deploy Remix + Vite anywhere?

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [remix-run/remix#8127] [Q&A] [answered]
Hi, 
I'm trying to deploy the Remix+Vite app and I can't get it to work. 
I've tried Vercel and Cloudflare Pages but both return 404 error after the deployment. 

I tried changing the build output directory to `public/build` as it seems to be the default output dir for this stack, but still no luck.

Is there any guide on how to deploy Remix/Vite app anywhere?

Thanks!

**Comment by gustavopch** [comment]
Your server must run `build/index.js` and you should serve static assets from the whole `public` folder, not only `public/build`.

Worth mentioning that these paths will change in a next release (https://github.com/remix-run/remix/pull/8077):
- The server build will be in `build/server` instead of `build`
- The client build will be in `build/client` instead of `public` (the `public` folder will still exist, but will be considered source files and will be copied to `build/client` when you build)

**Reply by tombohub** [reply]
so we run `/server/index.js` and set static folder to `build/client/` on our server?

**Comment by sergiodxa** [comment]
The Remix Vite plugin doesn't yet work on Cloudflare and Vercel, first remember that the Vite plugin is still unstable, so things like this are expected.

Cloudflare is not supported because Vite itself has issues with Cloudflare, the team is working on this.

Vercel is not supported because when deploying to Vercel they replace the `@remix-run/dev` package and force you to use their fork that has some changes they (Vercel team) did.

**Comment by hi-ogawa** [comment, marked as answer]
I expect Remix team has some plan to provide out-of-the-box experience in https://github.com/orgs/remix-run/projects/5/views/1?pane=issue&itemId=43815635 and https://github.com/orgs/remix-run/projects/5/views/1?pane=issue&itemId=43814919, but if you like to explore yourself, then I think underlying tools are available at least for simple deployment (not something like splitting server build https://github.com/remix-run/remix/pull/7118).

I made a quick demo to deploy Remix Vite app to Vercel Edge, Vercel Serverless, and Cloudflare Workers:
- https://github.com/hi-ogawa/remix-vite-deployment-examples/
- https://remix-vite-deploy-edge.vercel.app/
- https://remix-vite-deploy-serverless.vercel.app/
- https://remix-vite-deploy.hiro18181.workers.dev/

The idea is that, for Vercel, you can use `vercel deploy --prebuilt ...` to bypass vercel's out-of-the-box feature so that you can setup deployment "package" on your own by following their Build Output API https://vercel.com/docs/build-output-api/v3.

For cloudflare workers, I made a separate post in https://github.com/remix-run/remix/discussions/8099 to support workerd runtime api during development.

~(I'm not familiar with Cloudflare pages, so I didn't experiment it yet. I will take a look later if the same idea can be applied there.)~
(EDIT: I added Cloudflare Pages example https://remix-vite-deploy-pages.pages.dev/)

**Reply by johnantony92** [reply]
What was the vite configuration for the  cloudfare pages?.Also whats configured build path

**Comment by k-melnychuk** [comment]
Is there an example for Remix + Vite deployment on Netlify?

**Reply by w4t33r** [reply]
Hi, I hope my solution to this problem will help you.
- install  @remix-run/netlify
- create folder netlify/functions
- create server.mjs / js file
`import { createRequestHandler } from '@remix-run/netlify'`
`import * as build from '../../build/server/index.js'`
`
export const handler = createRequestHandler({
  build,
  getLoadContext() {
  },
});
`
- create netlify.toml [https://docs.netlify.com/configure-builds/file-based-configuration/](url)
`
[build]
command = "npm run build"
publish = "build/client"
functions = "netlify/functions"
`
`
[[redirects]]
from = "/*"
to = "/.netlify/functions/server"
status = 200
`

**Reply by devopsdina** [reply]
THANK YOU SO MUCH!!! I have been searching everywhere to find something clear like this.  All I wanted to do was serve up a simple blog, nothing crazy and the docs don't even show an example like this, nor is it in the templates for remix + vite + netlify.

Also worth noting the server.mjs/js file goes into the netlify/functions folder for anyone new.

**Comment by jonlambert** [comment]
To add to the original question – with Vite now stable, the story for deploying to Vercel is still pretty unclear. Would be great to understand the recommended approach 🙂

**Reply by tombohub** [reply]
@jonlambert vite remix is now stable?

**Reply by jonlambert** [reply]
https://github.com/remix-run/remix/pull/8713

Think it's just waiting on the release 😊

**Reply by tombohub** [reply]
Can we use express adapter and then deploy as express function?

**Comment by jasonsilberman** [comment]
Has anyone here ran into the "Error: Script startup exceeded CPU time limit" when deploying to Cloudflare? Currently hitting a hard wall debugging/trying to diagnose or find useful information online.

**Reply by naourass** [reply]
Might be the free plan cpu execution time limit. Paid plan offer (significantly) longer execution time I believe.

**Reply by tombohub** [reply]
after how many seconds it happens?

**Reply by tavoyne** [reply]
I do have the paid plan and am getting the same error.

**Reply by xkamail** [reply]
how to fix it ?
